modules/dnn/src/model.cpp

```diff
@@ -127,6 +127,49 @@ void ClassificationModel::classify(InputArray frame, int& classId, float& conf)
     std::tie(classId, conf) = classify(frame);
 }
 
+SegmentationModel::SegmentationModel(const String& model, const String& config)
+    : Model(model, config) {};
+
+SegmentationModel::SegmentationModel(const Net& network) : Model(network) {};
+
+Mat SegmentationModel::forward(InputArray frame){
+
+    std::vector<Mat> outs;
+    impl->predict(*this, frame.getMat(), outs);
+    CV_Assert(outs.size() == 1);
+    Mat score = outs[0];
+    const int batch = score.size[0];
```

Comment:
batch size should be `1` for a single frame, isn't it?

Reply:
Yes, I was just trying to make it generic in case the input has batch_size 2. Should I ignore this case?

Reply:
This method receives a single frame so batch_size == 1.